src/main/java/com/exactpro/th2/fix/client/FixMessage.java

```diff
@@ -24,6 +41,30 @@ public Object clone() {
         return message;
     }
 
+    public EventID getParentEventID() {
+        return parentEventID;
+    }
+
+    public void setParentEventID(EventID parentEventID) {
+        this.parentEventID = parentEventID;
+    }
+
+    public static ExpiringMap<String, EventID> getParentEventIDs() {
```

Comment:
```suggestion
    public static Map<String, EventID> getParentEventIDs() {
```
Also maybe replace it with a method which returns event-id for a certain message-string